src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1433,7 +1433,7 @@ def aks_agentpool_upgrade(cmd,
         instance.upgrade_settings.max_surge = max_surge
     if drain_timeout:
         instance.upgrade_settings.drain_timeout_in_minutes = drain_timeout
-    if node_soak_duration:
+    if node_soak_duration >= 0:
```

Comment:
>       if node_soak_duration >= 0:
E       TypeError: '>=' not supported between instances of 'NoneType' and 'int'

Some test failed.

I think you need to first check whether it is None or not
```suggestion
    if isinstance(node_soak_duration, int) and node_soak_duration >= 0:
```